airflow/providers/cncf/kubernetes/python_kubernetes_script.py

```diff
@@ -43,15 +43,22 @@ def remove_task_decorator(python_source: str, task_decorator_name: str) -> str:
 
     :param python_source:
     """
-    if task_decorator_name not in python_source:
-        return python_source
-    split = python_source.split(task_decorator_name)
-    before_decorator, after_decorator = split[0], split[1]
-    if after_decorator[0] == "(":
-        after_decorator = _balance_parens(after_decorator)
-    if after_decorator[0] == "\n":
-        after_decorator = after_decorator[1:]
-    return before_decorator + after_decorator
+
+    def _remove_task_decorator(py_source, decorator_name):
+        if decorator_name not in py_source:
+            return python_source
+        split = python_source.split(decorator_name)
+        before_decorator, after_decorator = split[0], split[1]
+        if after_decorator[0] == "(":
+            after_decorator = _balance_parens(after_decorator)
+        if after_decorator[0] == "\n":
+            after_decorator = after_decorator[1:]
+        return before_decorator + after_decorator
```

Comment:
```suggestion
        before_decorator, dec, after_decorator = python_source.partition(decorator_name)
        if not dec:
            return python_source
        if after_decorator.startswith("("):
            after_decorator = _balance_parens(after_decorator)
        after_decorator = after_decorator.lstrip("\n")
        return f"{before_decorator}{after_decorator}"
```

But

1. Why do we need to manunally balance parentheses?
2. The whitespace stripping logic seems a bit fragile to me
3. Is it possible to use `ast` for this?

Reply:
I would suggest we do a refactor in another PR instead of this one. Here I'm trying to include '@setup' and '@teardown' in the decorators to remove

Reply:
Yeah, I say we do refactoring separately.